### ISSUE TITLE
showLoading prop of SearchBar not working

### ISSUE THREAD

**Issue by Bradzer** [react-native-training/react-native-elements#1771]
> Before filling out an issue, be sure to check the version of react-native-elements you are using and the docs for that version over [here](https://react-native-training.github.io/react-native-elements/versions.html).

**Explain what you did**
I was trying to test the showLoading prop of the SearchBar by doing `showLoading= {true}`

**Expected behaviour**
I expected the activity indicator icon of the search bar to show

**Describe the bug**
The activity indicator icon didn't show up

**Your Environment (please complete the following information):**

| software              | version |
| --------------------- | ------- |
| react-native-elements |  1.0.0       |
| react-native          |   0.58.3      |
| npm or yarn           |    6.7.0     |

**Comment by iRoachie** [comment]
Seem to be working fine. Here's an example of all three search bars working with `showLoading`
https://snack.expo.io/@roach_iam/bad-cereal.

![kapture 2019-02-10 at 15 50 16](https://user-images.githubusercontent.com/5962998/52538656-9c531b00-2d4b-11e9-8647-089b5cfcd015.gif)

**Comment by Bradzer** [comment]
I think that it doesn't work because the `ActivityIndicator` of react native is buggy and doesn't work either. 
And I guess you built the prop on top of that `ActivityIndicator`

**Comment by iRoachie** [comment]
I've never had any issues with `ActivityIndicator`. `showLoading` uses `ActivityIndicator` directly https://github.com/react-native-training/react-native-elements/blob/master/src/searchbar/SearchBar-android.js#L126

**Comment by Bradzer** [comment]
Yes that's what I meant actually. You're using `ActivityIndicator` directly.
And it doesn't work because `ActivityIndicator` of react native doesn't work either.
A lot of people have experienced that `ActivityIndicator`.
I had to use another library myself to achieve my purpose.

**Comment by iRoachie** [comment]
I haven't come across anyone posting anything about `ActivityIndicator`. But nice to know.

**Comment by Bradzer** [comment]
You can check it out here https://github.com/facebook/react-native/issues/9023

**Comment by iRoachie** [comment]
That issue doesn't affect this lib. The `animating` prop is true by default and is not changed when you set the `showLoading` prop.

**Comment by Bradzer** [comment]
Still it doesn't work.
I noticed when showing the layouts bounds that when the `showLoading` is true the loading icon container is shown but the icon is still not displayed.

**Comment by iRoachie** [comment]
How are you using the SearchBar? Is it different from the example given above?

**Comment by Bradzer** [comment]
Here's my implementation : 
```
                <SearchBar 
                showLoading
                placeholder= 'Search specific word/expression...'
                value= {this.store.practiceSpecificWordSearch}
                onChangeText= {(changedText) => this.onSearchValueChanged(changedText)}
                onClear= {this.onSearchValueCleared}
                containerStyle={{marginBottom: 16}}
                returnKeyType='go'
                onSubmitEditing={this.onSearchSubmit}
                />
```
Here's a screenshot of the render : 
![screenshot_20190211-111824](https://user-images.githubusercontent.com/23164913/52580950-d847b800-2dff-11e9-9d74-9007aee1fe10.png)

As you can see, the icon container is there but the loading icon is still not visible.

**Comment by iRoachie** [comment]
Seems okay https://snack.expo.io/@roach_iam/amused-apple.

![kapture 2019-02-11 at 13 25 44](https://user-images.githubusercontent.com/5962998/52581218-8eab9d00-2e00-11e9-947a-ef032314d8ec.gif)

**Comment by Bradzer** [comment]
Just like you see there's nothing wrong with my code.
The bug of react native `ActivityIndicator` and this one are related.

**Comment by Bradzer** [comment]
Like I said I had to use another library that doesn't bug to achieve my purpose.
Here's the link https://github.com/joinspontaneous/react-native-loading-spinner-overlay
In case you might want to dive in and see how the problem has been solved over there because that library uses `ActivityIndicator`  of react native too.

**Comment by xavier-villelegier** [comment]
@Bradzer I couldn't reproduce this on my side neither. Did you try on a real device instead of a simulator, and on the latest RN release ? I can't see any related issues about the ActivityIndicator on the RN repo 🤔

**Comment by xavier-villelegier** [comment]
Closing this as inactive and we couldn't reproduce it, @Bradzer feel free to reopen it if you still have this issue

**Comment by pk936** [comment]
I am also facing the same issue. ShowLoading does not work.

**Comment by Juliocbr** [comment]
I have the same issue only for android

**Comment by dave-vazquez** [comment]
I was able to solve this issue with the following implementation:

```javascript
      <SearchBar
        lightTheme
        showLoading
        containerStyle={s.searchContainer}
        inputContainerStyle={s.inputContainer}
        loadingProps={{
          animating: searching,
          color: "black",
        }}
      />
```

I set animating in loadingProps to a boolean state variable, and to get the activity indicator to appear I just pass in a color that will show up with my custom styles.

Note that `animating` is set to true by default. You don't have to set it equal to any particular value, but if you want it to appear conditionally this would be the way to do it.
